src/popup-confirmation/tests/PopupConfirmation.spec.tsx

```diff
@@ -0,0 +1,161 @@
+const { describe, it } = intern.getInterface('bdd');
+const { assert } = intern.getPlugin('chai');
+import * as sinon from 'sinon';
+import { tsx } from '@dojo/framework/core/vdom';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import assertionTemplate from '@dojo/framework/testing/harness/assertionTemplate';
+import select from '@dojo/framework/testing/harness/support/selector';
+import { createHarness, compareTheme } from '../../common/tests/support/test-helpers';
+
+import Button from '../../button';
+import { PopupPosition } from '../../popup';
+import TriggerPopup from '../../trigger-popup';
+import PopupConfirmation from '../';
+
+import bundle from '../nls/PopupConfirmation';
+import * as css from '../../theme/default/popup-confirmation.m.css';
+
+const { messages } = bundle;
+const noop = () => {};
+const harness = createHarness([compareTheme]);
+
+describe('PopupConfirmation', () => {
+	const baseTemplate = assertionTemplate(() => (
+		<div classes={[css.root, undefined]}>
+			<TriggerPopup key="trigger-popup">
+				{{
+					trigger: noop as any,
+					content: noop as any
+				}}
+			</TriggerPopup>
+		</div>
+	));
+
+	const popupTemplate = (
```

Comment:
The preferred pattern is to create a single `baseTemplate` and then use setProperties / children etc to change the template rather than having template generator functions as they can get messy very quickly.